src/middleware/auth/logto-helper.ts

```diff
@@ -287,58 +258,104 @@ export class LogToHelper extends OAuthProvider implements IOAuthProvider {
 		}
 	}
 
-	private async getRoleInfo(roleId: string): Promise<ICommonErrorResponse> {
-		const uri = new URL(`/api/roles/${roleId}`, process.env.LOGTO_ENDPOINT);
-		try {
-			return await this.getToLogto(uri, 'GET');
-		} catch (err) {
-			return this.returnError(StatusCodes.BAD_GATEWAY, `getRoleInfo ${err}`);
-		}
-	}
-
-	private async assignDefaultRoleForUser(userId: string, roleId: string): Promise<ICommonErrorResponse> {
-		const userInfo = await this.getUserInfo(userId);
-		const uri = new URL(`/api/users/${userId}/roles`, process.env.LOGTO_ENDPOINT);
+	async assignCustomerPlanRoles(userId: string, planType: SupportedPlanTypes): Promise<ICommonErrorResponse> {
+		const [userInfo, userRoles] = await Promise.all([this.getUserInfo(userId), this.getRolesForUser(userId)]);
+		console.log('assignCustomerPlanRoles: userInfo: ', userInfo);
+		console.log('assignCustomerPlanRoles: userRoles: ', userRoles);
 
 		if (userInfo.status !== StatusCodes.OK) {
-			return this.returnError(
-				StatusCodes.BAD_GATEWAY,
-				`Could not fetch the info about role with roleId ${roleId}`
-			);
+			return this.returnError(StatusCodes.BAD_GATEWAY, 'Could not fetch the info about the user');
 		}
 		// Means that user exists
-		if (userInfo.data.isSuspended === 'true') {
+		if (userInfo.data.isSuspended) {
```

Comment:
Let's be mindful of the underlying type and / or possible coercion in effect here.

Otherwise, this might be an always true condition.

Reply:
I've checked it, this is a boolean. The existing condition would never has worked out and the reason why it went unnoticed is because there's no typings available on most of the responses in the project. It's an `any` here.